File: Wprowadzenie do funkcji/III. Wprowadzenie do funkcji/nww_nwd.py
def nww(liczba1, liczba2):
    liczba1_nww = liczba1
    liczba2_nww = liczba2
    while liczba1_nww != liczba2_nww:
        if liczba1_nww < liczba2_nww:
            liczba1_nww += liczba1
        else:
            liczba2_nww += liczba2

    return liczba1_nww


def nwd(liczba1, liczba2):
    wynik_nww = nww(liczba1, liczba2)
    wynik_nwd = (liczba1 * liczba2) / wynik_nww
    return int(wynik_nwd)


liczba1 = int(input("Podaj pierwszą liczbę "))
liczba2 = int(input("Podaj drugą liczbę "))
wynik_nww = nww(liczba1, liczba2)
wynik_nwd = nwd(liczba1, liczba2)
print(f"NWW({liczba1},{liczba2})={wynik_nww}")
print(f"NWD({liczba1},{liczba2})={wynik_nwd}")

File: Wprowadzenie do funkcji/IV. Wprowadzenie do funkcji/nww_nwd.cpp
#include <iostream>
using namespace std;

int nww(int liczba1, int liczba2)
{
    int liczba1Nww = liczba1;
    int liczba2Nww = liczba2;
    while(liczba1Nww != liczba2Nww)
    {
        if(liczba1Nww < liczba2Nww)
        {
            liczba1Nww += liczba1;
        }
        else
        {
            liczba2Nww += liczba2;
        }
    }

    return liczba1Nww;
}

int nwd(int liczba1, int liczba2)
{
    int wynikNww, wynikNwd;
    wynikNww = nww(liczba1, liczba2);
    wynikNwd = (liczba1*liczba2)/wynikNww;
    return wynikNwd;
}

int main()
{
    int liczba1, liczba2;
    int wynikNww, wynikNwd;
    cout << "Podaj dwie liczby naturalne ";
    cin >> liczba1 >> liczba2;
    wynikNww = nww(liczba1,liczba2);
    wynikNwd = nwd(liczba1,liczba2);
    cout << "NWW(" << liczba1 << "," << liczba2 << ")=" << wynikNww << endl;
    cout << "NWD(" << liczba1 << "," << liczba2 << ")=" << wynikNwd << endl;
    return 0;
}
